### PR TITLE
fixes in api and hatchery cli

### PR DESCRIPTION
**Pull request by proullon**

### CODE REVIEW

File: engine/hatchery/main.go [hunk 1]
```diff
@@ -59,9 +60,18 @@ func addCommands() {
...
 }
 
+// Cannot rely on viper.AutomaticEnv here because of the presence of hyphen '-'
+// in the flag (ie. for openstack cmd).
+// AutomaticEnv will only lookup for a strings.ToUpper
+// which will not replace hypen '-' with underscore '_'.
+// i.e "openstack-tenant" -> "CDS_OPENSTACK-TENANT" instead of "CDS_OPENSTACK_TENANT"
+// Hence SetEnvKeyReplacer.
+// Sadly, this isn't used by Flags().*Var functions, so no automatic binding on variables.
 func addFlags() {
```
Comment (fsamin) [inline engine/hatchery/main.go]: Oh nice ! i was stuck with it too !
is there a plan in viper to support/fix it ?

File: engine/hatchery/main.go [hunk 2]
```diff
@@ -32,11 +33,11 @@ var rootCmd = &cobra.Command{
 		}
 
 		if viper.GetString("api") == "" {
-			sdk.Exit("CDS api endpoint not provided. See help on flag --api\n")
+			sdk.Exit("CDS api endpoint [--api] not provided. See help on flag --api\n")
 		}
 
 		if viper.GetString("token") == "" {
-			sdk.Exit("Worker token not provided. See help on flag --token\n")
+			sdk.Exit("Worker token [--token] not provided. See help on flag --token\n")
```
Comment (proullon) [inline engine/hatchery/main.go]: yeah my bad, didn't read this far xD

File: engine/hatchery/main.go (hunk 1 above)
Comment (proullon) [inline engine/hatchery/main.go]: don't know, already spend 2h on this, I was too mad to fix it in viper

**Comment by proullon** [comment]
removed stuttering error message

**Comment by fsamin** [comment]
LGTM

**Comment by proullon** [comment]
can't merge, do it yourself \o

**Pull request merged**